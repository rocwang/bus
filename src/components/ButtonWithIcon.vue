<template>
  <button :class="$style.root" v-on="$listeners">
    <div :class="$style.icon">
      <slot name="icon" />
    </div>
    <span :class="$style.text">
      <slot name="text" />
    </span>
  </button>
</template>

<script>
export default {
  name: "ButtonWithIcon"
};
</script>

<style module>
.root {
  font-size: 1.6rem;
  text-align: center;
  padding: 7px 18px;
  background-color: var(--c-ocean-blue);
  color: var(--c-white);
  box-shadow: 2px 2px 4px 0 var(--c-black);
  display: flex;
  flex-flow: row nowrap;
  justify-content: flex-start;
  align-items: center;
}

.icon {
  width: 36px;
  margin-right: 18px;
}
</style>
