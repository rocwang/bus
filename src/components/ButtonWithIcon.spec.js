import { mount } from "@vue/test-utils";
import ButtonWithIcon from "./ButtonWithIcon";

describe("ButtonWithIcon.vue", () => {
  it("matches the snapshot", () => {
    const wrapper = mount(ButtonWithIcon);
    expect(wrapper.element).toMatchSnapshot();
  });
});
