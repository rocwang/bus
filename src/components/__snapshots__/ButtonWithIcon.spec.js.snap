// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonWithIcon.vue matches the snapshot 1`] = `
<button
  class="root"
>
  <div
    class="icon"
  />
   
  <span
    class="text"
  />
</button>
`;
